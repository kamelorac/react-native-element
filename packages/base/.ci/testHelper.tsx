import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
