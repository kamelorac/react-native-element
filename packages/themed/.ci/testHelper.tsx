import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
