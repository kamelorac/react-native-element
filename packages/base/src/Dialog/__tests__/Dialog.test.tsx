import React from 'react';
import { Text } from 'react-native';
import { Dialog } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('Dialog Component', () => {
  it('should render without any children', () => {
    const component = renderWithWrapper(<Dialog isVisible />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with title', () => {
    const { queryByText, queryByTestId } = renderWithWrapper(
      <Dialog isVisible>
        <Dialog.Title title="Dialog Title" />
        <Text testID="text">This is a test dialog</Text>
      </Dialog>
    );
    expect(queryByText('Dialog Title')).not.toBeNull();
    expect(queryByTestId('text').props.children).toBe('This is a test dialog');
  });

  it('should render buttons', () => {
    const { wrapper } = renderWithWrapper(
      <Dialog isVisible>
        <Text>This is a test dialog</Text>
        <Dialog.Actions>
          <Dialog.Button />
        </Dialog.Actions>
      </Dialog>,
      'Dialog__Button'
    );
    expect(wrapper).not.toBeNull();
  });

  it('should shows when isVisible is true', () => {
    const { wrapper } = renderWithWrapper(
      <Dialog isVisible>
        <Text>This is a test dialog</Text>
      </Dialog>,
      'Internal__Overlay'
    );
    // Modal Prop
    expect(wrapper.props.visible).toBeTruthy();
  });

  it('should not show when isVisible is false', () => {
    const { wrapper } = renderWithWrapper(
      <Dialog isVisible={false}>
        <Text>This is a test dialog</Text>
      </Dialog>,
      'Internal__Overlay'
    );
    expect(wrapper.props.visible).toBeFalsy();
  });
});
