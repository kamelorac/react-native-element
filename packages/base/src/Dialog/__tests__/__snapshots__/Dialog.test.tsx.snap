// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component should render without any children 1`] = `
<View
  testID="wrapper"
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    testID="Internal__Overlay"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, .4)",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="RNE__Overlay__backdrop"
    />
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
      testID="RNE__Overlay__Container"
    >
      <View
        style={
          {
            "backgroundColor": "rgb(0, 0, 0)",
            "borderRadius": 3,
            "padding": 20,
            "shadowColor": "rgba(0, 0, 0, .3)",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowRadius": 4,
            "width": "75%",
          }
        }
        testID="RNE__Overlay"
      />
    </View>
  </Modal>
</View>
`;
